client/components/error-boundary/index.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Notice } from '@wordpress/components';
+
+class ErrorBoundary extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			error: null,
+		};
+	}
+
+	static getDerivedStateFromError( error ) {
+		return { error };
+	}
+
+	componentDidCatch( error, info ) {
+		if ( this.props.onError ) {
+			this.props.onError( error, info );
+		}
+	}
+
+	render() {
+		if ( ! this.state.error ) {
+			return this.props.children;
+		}
+
+		return (
+			<Notice status="error" isDismissible={ false }>
+				{ __(
+					'There was an error rendering the component, please try again. Contact support for assistance if the problem persists.',
```

Comment:
I wouldn't mention "the component" in the user-facing error since it's kind of technical – maybe "this view"? Also "please try again" isn't necessarily actionable from the user's perspective, though I don't really have an alternative in mind.

Reply:
Makes sense, thanks! Copy updated. I've removed the "try again" part. I think it reads okay and doesn't give a possibly non-actionable instruction. Let me know if we better adjust it further!